### DISCUSSION TITLE
Developing custom rheology models?

### DISCUSSION THREAD

**Discussion by Aran55336** [chaos-polymtl/lethe#897] [Q&A]
Hello. Apologies for creating a discussion on something that has already been mentioned once before, but the links don't seem to work. I have seen on a previous discussion that it is possible to input rheology models into Lethe. How would I go about doing this? I cannot seem to find in the lethe files where they are stored and read that it is a fairly user friendly way of doing so.

**Comment by blaisb** [comment]
Dear Aran
All rheological models are implemented in the following class:
https://github.com/lethe-cfd/lethe/blob/master/source/core/rheological_model.cc
You can add a new rheological model in the .h and the .cc and make sure it is instantiated in the parser. Then you would have a new rheology model which would be compatible with the software itself.
Don't hesitate to make a pull request with your rheology model so we can merge it into the master branch
